packages/protocol/contracts/L1/TaikoToken.sol

```diff
@@ -2,18 +2,24 @@
 pragma solidity 0.8.24;
 
 import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
-import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20SnapshotUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20VotesUpgradeable.sol";
 import "../common/EssentialContract.sol";
 import "../common/LibStrings.sol";
 
+/// @notice TaikoToken was `EssentialContract, ERC20SnapshotUpgradeable, ERC20VotesUpgradeable`.
+/// We use this contract to take 50 more slots to remove `ERC20SnapshotUpgradeable` from the parent
+/// contract list.
+abstract contract EssentialContract_ is EssentialContract {
+    uint256[50] private __slots_previously_used_by_ERC20SnapshotUpgradeable;
+}
```

Comment:
Maybe a 'todo' here to dont forget to remove before mainnet, and only here bc. of testnet compatibility  ?

Reply:
added a comment, but not a TODO.